src/SubscribedStream.js

```diff
@@ -6,30 +6,60 @@ export default class SubscribedStream {
         this._client = client
         this.streamId = streamId
         this.subscriptions = {}
+        this.isPublisherPromises = {}
     }
 
-    getPublishers() {
+    async getPublishers() {
         if (!this.publishersPromise || (Date.now() - this.lastAccess) > PUBLISHERS_EXPIRATION_TIME) {
-            this.publishersPromise = this._client.getStreamPublishers(this.streamId)
+            this.publishersPromise = this._client.getStreamPublishers(this.streamId).then((publishers) => {
+                const map = {}
+                publishers.forEach((p) => {
+                    map[p] = true
+                })
+                return map
+            })
             this.lastAccess = Date.now()
         }
         return this.publishersPromise
     }
 
+    async isPublisher(publisherId) {
+        if (!this.isPublisherPromises[publisherId]) {
+            this.isPublisherPromises[publisherId] = this._client.isStreamPublisher(this.streamId, publisherId)
+        }
+        return this.isPublisherPromises[publisherId]
+    }
+
+    async isValidPublisher(publisherId) {
+        const cache = await this.getPublishers()
+        if (cache[publisherId]) {
```

Comment:
Isn't the code from here onwards in this function basically the same logic as what `isPublisher` does? As `cache == this.isPublisherPromises`?

Reply:
`cache == this.publishersPromise` but `cache != this.isPublisherPromises`.
The logic is different: it uses the first endpoint to get the full list and periodically refresh it and uses the second endpoint when there is a cache miss.

Reply:
Ah my bad. I mixed them up and thought there is only one variable.